diff -Nru ffmpeg-3.4.6-01-orig/libavdevice/Makefile ffmpeg-3.4.6-02-fix-v4l2/libavdevice/Makefile
--- ffmpeg-3.4.6-01-orig/libavdevice/Makefile	2019-03-29 02:49:18.000000000 +0200
+++ ffmpeg-3.4.6-02-fix-v4l2/libavdevice/Makefile	2020-01-18 02:32:55.857083408 +0200
@@ -7,6 +7,7 @@
 OBJS    = alldevices.o                                                  \
           avdevice.o                                                    \
           utils.o                                                       \
+          v4l2.o v4l2-common.o timefilter.o				\
 
 OBJS-$(HAVE_LIBC_MSVCRT)                 += file_open.o
 OBJS-$(CONFIG_SHARED)                    += reverse.o
