diff -Nru linux-4.14.75-02-mxc/arch/arm/boot/dts/imx6ul.dtsi linux-4.14.75-03-csi-dts/arch/arm/boot/dts/imx6ul.dtsi
--- linux-4.14.75-02-mxc/arch/arm/boot/dts/imx6ul.dtsi	2018-10-10 09:54:28.000000000 +0300
+++ linux-4.14.75-03-csi-dts/arch/arm/boot/dts/imx6ul.dtsi	2020-01-17 02:24:54.508234635 +0200
@@ -866,6 +866,17 @@
 				clocks = <&clks IMX6UL_CLK_OCOTP>;
 			};
 
+			csi: csi@021c4000 {
+				compatible = "fsl,imx6ul-csi", "fsl,imx6s-csi";
+				reg = <0x021c4000 0x4000>;
+				interrupts = <GIC_SPI 7 IRQ_TYPE_LEVEL_HIGH>;
+				clocks = <&clks IMX6UL_CLK_DUMMY>,
+					<&clks IMX6UL_CLK_CSI>,
+					<&clks IMX6UL_CLK_DUMMY>;
+				clock-names = "disp-axi", "csi_mclk", "disp_dcic";
+				status = "disabled";
+			};
+
 			lcdif: lcdif@021c8000 {
 				compatible = "fsl,imx6ul-lcdif", "fsl,imx28-lcdif";
 				reg = <0x021c8000 0x4000>;
