--- a/drivers/net/usb/qmi_wwan.c
+++ b/drivers/net/usb/qmi_wwan.c
@@ -1011,7 +1011,7 @@ static const struct usb_device_id produc
 	{QMI_FIXED_INTF(0x05c6, 0x7102, 3)},
 	{QMI_FIXED_INTF(0x05c6, 0x8000, 7)},
 	{QMI_FIXED_INTF(0x05c6, 0x8001, 6)},
-	{QMI_FIXED_INTF(0x05c6, 0x9000, 4)},
+	{QMI_QUIRK_SET_DTR(0x05c6, 0x9000, 4)},
 	{QMI_FIXED_INTF(0x05c6, 0x9003, 4)},
 	{QMI_FIXED_INTF(0x05c6, 0x9005, 2)},
 	{QMI_FIXED_INTF(0x05c6, 0x900a, 4)},
